### PR TITLE
Added `Tree` component

### LINKED ISSUE #990: Add `Tree`
### Description

Trees allow you to display a hierarchical list of selectable tree items. Items with children can be expanded and collapsed as desired by the user.

- [Tree](https://shoelace.style/components/tree)

### Problem Statement/Justification

Depending on your application, you may want to display data like a tree.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by hirotomoyamada**
Closes #990

## AI used

- [x] I did not use AI to create this PR.
- [ ] (If there is no check above) I checked the generated content before submitting.

## Description

Added `Tree` component.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/react/src/components/tree/use-tree.ts [hunk 1]
```diff
@@ -0,0 +1,932 @@
...
+  const onGroupOpen = useCallback(() => {
+    if (disabled || !value) return
+
+    if (async && !initialAsyncRef.current) {
+      initialAsyncRef.current = true
+
+      onGroupChildren().then(() => {
+        onOpen()
+        onExpandedChange(value)
+      })
```
[the remark this answers is not in the document]
Comment (hirotomoyamada) [reply packages/react/src/components/tree/use-tree.ts:555]: Fixed.

**Review by claude-for-yamada-ui — APPROVED** [review]
LGTM! Well-structured `Tree` component with comprehensive features:

- Clean component architecture with `TreeRoot`, `TreeItem`, and supporting subcomponents
- Full keyboard navigation support (Arrow keys, Home/End, Enter/Space, Ctrl+A, Shift+selection)
- Proper ARIA attributes (`role="tree"`, `aria-expanded`, `aria-activedescendant`, `aria-level`, etc.)
- Async children loading with loading state
- Checkbox support with parent-child propagation and indeterminate state
- Multiple selection with Ctrl+Click and Shift+Click range selection
- Guide lines, custom indicators, start/end elements
- The `useDescendant` fix to update props on re-render is correctly scoped and separately changesetted
- Thorough test coverage (966 lines) covering a11y, interactions, keyboard, async, and edge cases
